### DISCUSSION TITLE
How to get Prettier to run

### DISCUSSION THREAD

**Discussion by Kay2dan** [blitz-js/blitz#948] [Q&A] [answered]
I have just started out with Blitz and while my editor (vscode) is already configured & previously used with Prettier, however, I cannot seem to get the Blitz code to format by prettier alongside eslint. 

I have already created & setup my .prettierrc file in the root directory. How do you guys make them run together.

**Comment by flybayer** [comment, marked as answer]
Hey @Kay2dan, I don't use VSCode, so not sure what's the issue here. Hopefully someone else with VSCode can comment here.

However, one thing to note is that Blitz apps have a default prettier config in package.json. So I don't know which config will actually take affect, package.json or .prettierrc

**Reply by Kay2dan** [reply]
Thank you so much! That was it, I hadn't noticed the settings within the `package.json` & once I deleted it, my `.prettierrc` is being picked up now. 

And thank you for Blitz, its a bit intimidating to get into, however, it looks very neat & exciting.

**Reply by flybayer** [reply]
Yay! Happy to help. Let us know if we can help any more. You can also join our slack at [slack.blitzjs.com](https://slack.blitzjs.com)

**Comment by silicontwin** [comment]
I also had this issue, and tried the solution proposed by @Kay2dan, but it didn't work for me. What did work was adding the following line to the `settings.json` file inside the `.vscode` directory that ships with a fresh Blitz.js install: `"editor.defaultFormatter": "esbenp.prettier-vscode",`. This line, "ensures that this extension is used over other extensions you may have installed." More info [here](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode). Hope this helps others in the future. Great work Blitz.js team!

**Reply by flybayer** [reply]
Thanks! Should we add that line by default?

**Reply by silicontwin** [reply]
Happy to help! Adding that line by default could be a good idea to help avoid additional issues in the future. The line I added applies to all languages, but the Prettier page I linked has info on configuring that line to only apply to JavaScript, etc.

Since the line scopes the setting to the user's project directory (as I understand it), it probably wouldn't be an issue—though it would be a good idea to test with some people in advance.

**Reply by flybayer** [reply]
Ok yes I agree, leave it for all languages. Here's the file that needs updated: https://github.com/blitz-js/blitz/blob/canary/packages/generator/templates/app/.vscode/settings.json

**Reply by silicontwin** [reply]
Just added a PR with the line added to that file.
